libs/cardano-ledger-core/cddl/Cardano/Ledger/Core/HuddleSpec.hs

```diff
@@ -246,29 +266,27 @@ instance Era era => HuddleRule "script_hash" era where
           |  "\x03" for Plutus V3 scripts
           |  "\x04" for Plutus V4 scripts
           |]
-      $ "script_hash"
-        =:= huddleRule @"hash28" p
 
 instance Era era => HuddleRule "credential" era where
-  huddleRule p =
-    "credential"
-      =:= arr [0, a (huddleRule @"addr_keyhash" p)]
-      / arr [1, a (huddleRule @"script_hash" p)]
+  huddleRuleBody p =
+    Type0 $
+      arr [0, a (huddleRule @"addr_keyhash" p)]
+        / arr [1, a (huddleRule @"script_hash" p)]
 
 instance Era era => HuddleRule "stake_credential" era where
-  huddleRule p = "stake_credential" =:= huddleRule @"credential" p
+  huddleRuleBody p = Type0 $ huddleRule @"credential" p
 
 instance Era era => HuddleRule "port" era where
-  huddleRule _ = "port" =:= VUInt `le` 65535
+  huddleRuleBody _ = Type0 $ VUInt `le` 65535
 
 instance Era era => HuddleRule "ipv4" era where
-  huddleRule _ = "ipv4" =:= VBytes `sized` (4 :: Word64)
+  huddleRuleBody _ = Type0 $ VBytes `sized` (4 :: Word64)
 
 instance Era era => HuddleRule "ipv6" era where
-  huddleRule _ = "ipv6" =:= VBytes `sized` (16 :: Word64)
+  huddleRuleBody _ = Type0 $ VBytes `sized` (16 :: Word64)
 
-majorProtocolVersionRule :: forall era. Era era => Proxy era -> Rule
+majorProtocolVersionRule :: forall era. Era era => Proxy era -> Type0
 majorProtocolVersionRule _ =
-  "major_protocol_version"
-    =:= getVersion @Integer (eraProtVerLow @ByronEra)
-    ... succ (getVersion @Integer (eraProtVerHigh @era))
+  Type0 $
```

Comment:
If you create a helper that is called `toType0 = Type0`, then transitioning to new version of cuddle that exports `Type0` type synonym will have a much smaller impact.
```suggestion
  toType0 $
```